src/renderer/components/composer/ComposerMessageInput.tsx

```diff
@@ -28,14 +30,44 @@ export default class ComposerMessageInput extends React.Component<
   setCursorPosition: number | false
   textareaRef: React.RefObject<HTMLTextAreaElement>
   saveDraft: () => void
+  recorder: MediaRecorder | null = null
+  updateRecordedSecondsInterval: number | null = null
+  voiceData: Blob[] = []
+  gumError: any
   constructor(props: ComposerMessageInputProps) {
     super(props)
     this.state = {
       text: '',
       chatId: props.chatId,
       loadingDraft: true,
+      recordedDuration: null,
     }
-
+    navigator.mediaDevices.getUserMedia({ audio: true, video: false }).then(stream => {
```

Comment:
I have a feeling we should do the setup only if the user clicked on record. also we might need a runtime call to check and request permission?

Also later if could make some sense to ask user for which microphone they wanna use If they have multiple ones connected, or we somehow always take the default?

Reply:
For now, I'd say merge the PR with the default microphone. For later, we could add something into Settings.

Reply:
I'm not sure if we should only request permission when recording starts. What is your reason? TBH, I don't see any difference.

Reply:
> What is your reason?

apple design guidelines and other places, it's best practice to ask for permission when its needed.
also sometimes access to microphone is shown similar to how most webcams have a light when they are recording.
If that indication is on when they don't record voice messages we will get complains and new conspiracy theories.

I mean get the stream only when you click on the button and use it

Reply:
Yeah it makes perfect sense. Will change it and resolve this convo